build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/jsoncdb/generator/CompilationDatabaseGenerator.java

```diff
@@ -438,15 +438,15 @@ public boolean visit(IResourceProxy proxy) throws CoreException {
 
 	}
 
-	private String findCompilerInPath(ITool tool, IConfiguration config) {
-		if (toolMap.containsKey(tool)) {
-			return "\"" + toolMap.get(tool) + "\""; //$NON-NLS-1$//$NON-NLS-2$
+	private String findCompilerInPath(ITool tool, IConfiguration config, String compilerName) {
+		String cacheKey = tool + " " + compilerName; //$NON-NLS-1$
```

Comment:
The `toString` of ITool is just the externalized name of the tool, not really suitable for using as a key here.

That said and on further examination, we only need to consider `tool` for the cache key if the compilerName is not unique enough. AFAICT it is unique with a compilation, so probably easiest to just drop tool entirely here.

```suggestion
		String cacheKey = compilerName; //$NON-NLS-1$
```